deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,280 @@
+        if self.use_dropout:
+            dtnn_step = nn.Dropout(self.dropout)(dtnn_step)
+        dtnn_gather = self.dtnn_gather([dtnn_step, inputs[2]])
+        if self.use_dropout:
+            dtnn_gather = nn.Dropout(self.dropout)(dtnn_gather)
+        output = self.linear(dtnn_gather)
+        return output
+
+
+class DTNNModel(TorchModel):
+    """Implements DTNN models for regression.
+
+    This class implements the Directed Message Passing Neural Network (D-MPNN) [1]_.
+
+    Examples
+    --------
+    >>> import deepchem as dc
+    >>> import os
+    >>> from scipy import io as scipy_io
+    >>> from deepchem.data import NumpyDataset
+    >>> model_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    >>> input_file = os.path.join(model_dir, 'tests/assets/example_DTNN.mat')
+    >>> dataset = scipy_io.loadmat(input_file)
+    >>> X = dataset['X']
+    >>> y = dataset['T']
+    >>> w = np.ones_like(y)
+    >>> dataset = NumpyDataset(X, y, w, ids=None)
+    >>> n_tasks = y.shape[1]
+    >>> model = DTNNModel(n_tasks,
+    ...               n_embedding=20,
+    ...               n_distance=100,
+    ...               learning_rate=1.0,
+    ...               mode="regression")
+    >>> loss = model.fit(dataset, nb_epoch=250)
+    >>> pred = model.predict(dataset)
+
+    References
+    ----------
+    .. [1] Schütt, Kristof T., et al. "Quantum-chemical insights from deep
+        tensor neural networks." Nature communications 8.1 (2017): 1-8.
+
+    """
+
+    def __init__(self,
+                 n_tasks: int,
+                 n_embedding: int = 30,
+                 n_hidden: int = 100,
+                 n_distance: int = 100,
+                 distance_min: float = -1,
+                 distance_max: float = 18,
+                 output_activation: bool = True,
+                 mode: str = "regression",
+                 dropout: float = 0.0,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        n_embedding: int (default 30)
+            Number of features per atom.
+        n_hidden: int (default 100)
+            Number of features for each molecule after DTNNStep
+        n_distance: int (default 100)
+            granularity of distance matrix
+            step size will be (distance_max-distance_min)/n_distance
+        distance_min: float (default -1)
+            minimum distance of atom pairs (in Angstrom)
+        distance_max: float (default = 18)
+            maximum distance of atom pairs (in Angstrom)
+        output_activation: bool (default True)
+            determines whether an activation function should be apply  to its output.
+        mode: str (default "regression")
+            Only "regression" is currently supported.
+        dropout: float (default 0.0)
+            the dropout probablity to use.
+
+        """
+        if dropout < 0 or dropout > 1:
```

Comment:
I see that you have validated it here itself, then just remove use_dropout and all the if checks for self.dropout > 0.0, as this is really unnecessary and nn.dropout supports 0 probability

Reply:
I was not knowing that, thanks.
Implementing.